engine/lib/entities.php

```diff
@@ -677,6 +678,43 @@ function elgg_search(array $options = []) {
 	}
 }
 
+/**
+ * Return an array reporting the number of various entities in the system.
+ *
+ * @param int $owner_guid Optional owner of the statistics
+ *
+ * @return array
+ */
+function get_entity_statistics($owner_guid = 0) {
+
+	$owner_guid = (int) $owner_guid;
+	$entity_stats = [];
+
+	$grouped_entities = elgg_get_entities([
+		'selects' => ['COUNT(*) as cnt'],
+		'owner_guids' => ($owner_guid) ? : ELGG_ENTITIES_ANY_VALUE,
+		'group_by' => ['e.type', 'e.subtype'],
+		'limit' => 0,
+		'order_by' => new OrderByClause('cnt', 'DESC'),
```

Comment:
wouldn't it be wiser to have `callback => false` and use the database row to work with instead of first converting it to an ElggEntity

Reply:
will change this